### ISSUE TITLE
Dependency Conflict with netinfo with latest Expo version

### ISSUE THREAD

**Issue by VirenMohindra** [magiclabs/magic-js#676]
### ✅ Prerequisites

- [x] Did you perform a cursory search of open issues? Is this bug already reported elsewhere?
- [x] Are you running the latest SDK version?
- [x] Are you reporting to the correct repository (`magic-sdk`)?

### 🐛 Description

The latest Expo version expects`@react-native-community/netinfo` to be 9.3.10, but `@magic-sdk/react-native-expo` expects any version above 11.0.0.

This is flagged while running `npm i` as well as `npx expo-doctor`, and is causing instability in our app.

### 🧩 Steps to Reproduce

1. Create an expo project
2. Install magic with required peer dependencies
3. Run `npx expo-doctor`

### 🤔 Expected behavior

I would expect no errors and no peer dependency issues.

### 💻 Error Output from Terminal

```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: gently-native@2.3.0
npm ERR! Found: @react-native-community/netinfo@9.3.10
npm ERR! node_modules/@react-native-community/netinfo
npm ERR!   @react-native-community/netinfo@"9.3.10" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer @react-native-community/netinfo@">11.0.0" from @magic-sdk/react-native-expo@22.3.2
npm ERR! node_modules/@magic-sdk/react-native-expo
npm ERR!   @magic-sdk/react-native-expo@"^22.3.2" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! 
npm ERR! For a full report see:
npm ERR! /.npm/_logs/2023-11-29T00_08_32_328Z-eresolve-report.txt
  ```

In this code sample, the last `console.log()` never returns anything.

### 🌎 Environment

| Software         | Version(s) | Comments
| ---------------- | ---------- | ---------- |
| `@magic-sdk/react-native-expo`      | 22.3.2 | latest magic-sdk version
| `@react-native-community/netinfo`      | 9.3.10 | netinfo version that expo expects
| `expo` | ~49.0.13 | latest Expo version
| `react-native` | 0.72.6

**Comment by VirenMohindra** [comment]
another issue - `@react-native-community/netinfo` latest version 11.1.1 **does not** work on Expo. so not only is there a peer dependency issue, its just flat out broken. the modal does not show up.

**Comment by romin-halltari** [comment]
Thank you for reporting this @VirenMohindra, will submit a new version with the fix today.

**Comment by romin-halltari** [comment]
@VirenMohindra The fix has been released on version `22.3.4`. Please let us know if any other issues arise.

**Comment by VirenMohindra** [comment]
Dependency conflict seems to be fixed. `"expo": "~49.0.13",` works with `"@react-native-community/netinfo": "9.3.10"`
